### PR TITLE
#164198172 Setup DB engine configuration with PostgresSQL

### PR DESCRIPTION
**Pull request by devdbrandy**
#### What does this PR do?

Configure db engine configuration with PostgresSQL

#### Description of Task to be completed

- [x] Install relevant dependencies in conjunction with Sequelize ORM
- [x] Configure sequlize to use PostgresSQL as the database engine
- [x] Setup migration and seeders
- [x] Refactor implementation to meet es6 standard
- [x] Configure travisCI to use postgres service
- [ ] Refactor models to use sequelize pattern of model declaration

#### How should this be manually tested

_Prerequisite: Postgres_

1. Run the following commands on terminal:

```bash
git clone -b chore/164198172/configure-postgres --single-branch \
https://github.com/andela/apollo-ah-backend.git
cd apollo-ah-backend
cp .env.example .env
npm install
```

2. Modify `.env` file with your db credentials, then run any of the provided script commands on terminal.

- `npm run db:create` - Creates the database.

- `npm run make:model <name> --attributes <attribute>:<type>` - Generates model, where `name` is the name of the model, `attribute` is the attribute of the model and `type` is the attribute data type.

- `npm run migrate`: Runs the migration.

#### Any background context you want to provide

Documentation on how to go about using the cli tool to generate models, migrations, seeders will be provided on the Wiki page.
**[Update]** Documentation is available on [Wiki](https://github.com/andela/apollo-ah-backend/wiki/Guides:-Database-Models,-Migrations-and-Seeding) page.

#### What are the relevant pivotal tracker stories

[#164198172](https://www.pivotaltracker.com/story/show/164198172)

#### Screenshots

n/a

### CODE REVIEW

**Comment by aanchirinah** [comment]
@devdbrandy , pull recent changes and resolve this.

File: models/index.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+};
+
+Object.keys(models).forEach((key) => {
+  if (models[key].associate) models[key].associate(models);
+});
+
+models.sequelize = sequelize;
+models.Sequelize = Sequelize;
+
+export default models;
```
[the remark this answers is not in the document]
Comment (devdbrandy) [reply models/index.js]: Hound config setup #4 is dependant on PR #5. Hound checks is based on its default configuration 'esversion: 6'.

screen shot 2019-02-26 at 1 45 27 pm

**Comment by devdbrandy** [comment]
@aanchirinah I have introduced the recent changes, conflict resolved,  thanks.

**Comment by devdbrandy** [comment]
@aanchirinah PR is ready for review.

File: config/database.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ * Sequelize config options
+ */
+const options = {
+  logging: env('DB_LOGGING', false), // disable query logging
+  operatorsAliases: false, // Optional: disable deprecation warning
+};
+
+export default {
+  development: getConfig(options),
+  test: getConfig(options),
```
Comment (aanchirinah) [inline config/database.js]: Can you please setup an external DB for testing to not break Travis CI/CD

**Comment by aanchirinah** [comment]
Also you have conflicts , please pull recent changes. You are doing an awesome so far so I have confidence you can do more.

File: config/database.js (hunk 2 above)
Comment (devdbrandy) [reply config/database.js]: Will work on it.

Comment (devdbrandy) [reply config/database.js]: Issue resolved 👍

**Pull request merged**